src/clj/puppetlabs/http/client/async.clj

```diff
@@ -192,16 +191,18 @@
   [opts http-response]
   (let [headers       (get-resp-headers http-response)
         orig-encoding (headers "content-encoding")]
-    {:opts                  opts
+    {:opts opts
      :orig-content-encoding orig-encoding
-     :status                (.. http-response getStatusLine getStatusCode)
-     :headers               headers
-     :content-type          (parse-content-type (headers "content-type"))
-     :body                  (do
-		              (if (:decompress-body opts)
-		                (.process (ResponseContentEncoding.) http-response nil))
-		              (when-let [entity (.getEntity http-response)]
-		                (.getContent entity)))}))
+     :status (.. http-response getStatusLine getStatusCode)
+     :headers headers
+     :content-type (parse-content-type (headers "content-type"))
+     :body (do
+             (if (:decompress-body opts)
+               (.process (ResponseContentEncoding.)
+                         http-response
+                         (BasicHttpContext.)))
```

Comment:
```ResponseContentEncoding``` came in when I added the ```:unbuffered-stream``` stuff. I did that only on the Clojure side to try to hit a deadline - previously the Clojure and Java approach were analogous. I think it's on me to resolve the inconsistency there.